lib/module-types.js

```diff
@@ -0,0 +1,51 @@
+module.exports = (configuredExtensions, defaultModuleType) => {
+	const lockedModuleTypes = {
+		cjs: 'commonjs',
+		mjs: 'module',
+		js: defaultModuleType
+	};
+
+	const createModuleTypesFromExtensionsObject = extensionsObject => {
+		const configuredModuleTypes = {};
+
+		for (const [extension, type] of Object.entries(extensionsObject)) {
+			if (extension in lockedModuleTypes && type !== lockedModuleTypes[extension]) {
+				throw new TypeError(`.${extension} files can only be configured as ’${lockedModuleTypes[extension]}’${
+					extension === 'js' ?
+						` when the nearest parent package.json ${type === 'module' ? 'does not contain' : 'contains'} "type": "module"` : ''
+				}, found ’${type}’ instead.`);
+			}
+
+			configuredModuleTypes[extension] = type;
+		}
+
+		return configuredModuleTypes;
+	};
+
+	const createModuleTypesFromExtensionsArray = extensions => {
+		const generatedModuleTypes = {};
+
+		for (const extension of extensions) {
+			if (extension in lockedModuleTypes) {
+				generatedModuleTypes[extension] = lockedModuleTypes[extension];
+			} else {
+				generatedModuleTypes[extension] = 'commonjs';
```

Comment:
This is kinda useless for now, I know, but in the future _unknown_ extensions (`.ts`, `.re`, etc) could be loaded with the default type, just like `.js` is. Doing that would be as easy as changing `"commonjs"` to `defaultModuleType`. That's gonna be a breaking change tho.